core/shared/src/main/scala/com/lynbrookrobotics/potassium/streams/Stream.scala

```diff
@@ -508,35 +385,20 @@ abstract class Stream[+T] { self =>
       val cancel = self.foreach(v => updateEventState(condition.apply(v)))
     }
   }
-
-  /**
-    * Adds a listener for elements of this stream. Callbacks will be executed
-    * whenever a new value is published in order of when the callbacks were added.
-    * Callbacks added first will be called first and callbacks added last will
-    * be called last.
-    *
-    * @param thunk the listener to be called on each published value
-    * @return a function to call to remove the listener
-    */
-  def foreach(thunk: T => Unit): Cancel = {
-    self.synchronized {
-      listeners = listeners :+ thunk
-    }
-
-    () => {
-      self.synchronized {
-        listeners = listeners.filterNot(_ eq thunk)
-      }
-    }
-  }
 }
 
 object Stream {
   def periodic[T](period: Time)(value: => T)(implicit clock: Clock): Stream[T] = {
     new Stream[T] { self =>
```

Comment:
Doesn't this whole thing need to be persistent?

Reply:
See below!